### PR TITLE
[bitnami/wordpress] MAJOR: WordPress image refactor

### PR DESCRIPTION
**Pull request by marcosbc**
**Description of the change**

Preventive major bump of WordPress chart + Add support for new features

**Benefits**

Users can enjoy an improved container image :) 

**Possible drawbacks**

Upgrades are not guaranteed to work

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

**Review by javsalgar — APPROVED** [review]
The PR overall looks good to me. I would clarify that the multisite options will only work on first boot.

File: bitnami/wordpress/templates/deployment.yaml [hunk 1]
```diff
@@ -145,20 +145,36 @@ spec:
...
             - name: WORDPRESS_TABLE_PREFIX
               value: {{ .Values.wordpressTablePrefix | quote }}
             - name: WORDPRESS_SCHEME
               value: {{ .Values.wordpressScheme | quote }}
             - name: WORDPRESS_EXTRA_WP_CONFIG_CONTENT
               value: {{ .Values.wordpressExtraConfigContent | quote }}
+            - name: WORDPRESS_AUTO_UPDATE_LEVEL
+              value: {{ .Values.wordpressAutoUpdateLevel | quote }}
+            - name: WORDPRESS_PLUGINS
+              value: {{ .Values.wordpressPlugins | quote }}
```
Comment (juan131) [inline bitnami/wordpress/templates/deployment.yaml:163]: I'd use an array format for `wordpressPlugins`. Then, you can use:
```suggestion
              value: {{ join "," .Values.wordpressPlugins | quote }}
```

File: bitnami/wordpress/values.yaml [hunk 2]
```diff
@@ -131,6 +131,13 @@ existingWordPressConfigurationSecret:
 ## NOTE: useful if you deploy Memcached for caching database queries or you use an external cache server
 ##
 wordpressConfigureCache: false
+## @param wordpressAutoUpdateLevel Level of auto-updates to allow. Allowed values: `major`, `minor` or `none`.
+##
+wordpressAutoUpdateLevel: none
+## @param wordpressPlugins Plugins to install and activate, comma-separated. Can also be set to `all` or `none`.
+## NOTE: If set to all, only plugins that are already installed will be activated, and if set to none, no plugins will be activated
+##
+wordpressPlugins: none
```
Comment (juan131) [inline bitnami/wordpress/values.yaml:140]: I wouldn't use a "comma-separated" format, but an array instead. In addition to this, shouldn't be possible to indicate what plugins to install and what plugins to activate separately?

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks for the great PR! Please check my comments

File: bitnami/wordpress/values.yaml (hunk 2 above)
Comment (marcosbc) [reply bitnami/wordpress/values.yaml:140]: Well, I guess it should be possible to allow for both arrays, comma-separated and strings (like `none` or `all`). I'll update the description.

Comment (marcosbc) [reply bitnami/wordpress/values.yaml]: BTW I don't think adding complexity to the users (with respect to having the option to specify plugins to activate and install separately) is something we should be doing, at least in the first iteration of these changes. I think it is prone to cause more confusion than good.

Comment (juan131) [reply bitnami/wordpress/values.yaml]: > Well, I guess it should be possible to allow for both arrays, comma-separated and strings (like none or all)

It seems this new approach to support both a string and an array works:

```console
$ values="$(cat << EOF
wordpressPlugins:
  - foo
  - bar
EOF
)"
$ helm template wordpress bitnami/wordpress -f <(echo "$values") -s templates/deployment.yaml
(...)
            - name: WORDPRESS_PLUGINS
              value: "foo,bar"
$ helm template wordpress bitnami/wordpress -s templates/deployment.yaml
(...)
            - name: WORDPRESS_PLUGINS
              value: "none"
```

**Review by juan131 — APPROVED** [review]
LGTM

**Pull request merged**